skimage/feature/_kb.py

```diff
@@ -0,0 +1,264 @@
+from .._shared.utils import assert_nD
+
+
+def kb(image, min_scale=10, max_scale=25, saliency_threshold=0.6, clustering_threshold=7):
+    """Finds salient regions in the given grayscale image.For each point x
+    the method picks a scale s and calculates salient score Y(x,s).
+    By comparing Y(x,s) of different points x the detector can rank
+    the saliency of points and pick the most representative ones.
+
+    Parameters
+    ----------
+    image : ndarray
+        Input grayscale image
+    min_scale : int, optional
+        Minimum scale for the keypoints detected.
+    max_scale : int, optional
+        Maximum scale for the keypoints detected.
+    saliency_threshold : float, optional.
+        Features with saliency score satisfying threshold will be considered.
+    clustering_threshold : int, optional
+        Variance among the regions should be smaller than this threshold
+        for sufficient clustering.
+
+    Returns
+    -------
+    A : (n, 3) ndarray
+        A 2d array with each row representing 3 values, '(y, x, scale)'
+        where '(y,x)' are coordinates of the region and 'scale' is the
+        size of corresponding salient region.
+
+    References
+    ----------
+    https://en.wikipedia.org/wiki/Kadir-Brady_saliency_detector
+
+    Examples
+    --------
+    >>> from skimage.feature import kb
+    >>> from skimage.color import rgb2gray
+    >>> from skimage.data import astronaut
+    >>> image = astronaut()[100:300, 100:300]
+    >>> kb(rgb2gray(image))
+    array([[   0.875,  102.625,   12.75 ],
+           [ 127.875,   72.125,   12.   ],
+           [   1.875,  146.125,   12.   ],
+           [ 163.875,  149.875,   14.   ],
+           [  14.875,   75.125,   14.   ],
+           [ 155.125,  105.875,   12.   ],
+           [  92.75 ,   46.625,   14.   ],
+           [ 159.625,  127.25 ,   18.25 ],
+           [ 107.625,   16.5  ,   12.25 ],
+           [   0.875,  191.875,   14.5  ],
+           [  45.625,  142.125,   12.5  ],
+           [ 137.875,  192.   ,   12.25 ],
+           [ 106.625,   59.125,   13.   ],
+           [  91.375,   18.5  ,   13.25 ],
+           [ 152.   ,  184.125,   12.5  ],
+           [ 149.125,   91.125,   12.   ],
+           [ 122.75 ,   28.875,   12.75 ],
+           [  82.625,  120.   ,   13.75 ],
+           [  95.875,   30.75 ,   12.25 ],
+           [  16.125,  165.25 ,   15.75 ],
+           [  57.875,   76.375,   23.5  ],
+           [  86.125,   56.5  ,   12.   ],
+           [  80.75 ,   29.875,   12.5  ]])
+
+    Notes
+    -----
+    The radius of each region is 'scale/2'.
+    """
+
+    assert_nD(image, 2)
+
+    # scales for keypoints
+    scales = np.array([_ for _ in range(min_scale, max_scale, 2)])
+    # detect keypoints in scale-space
+    base_regions = detect(image, scales)
+    # pruning based on thresholds
+    regions = prune(base_regions, saliency_threshold, clustering_threshold, K=7)
+
+    return regions.T
+
+
+def detect(image, scales):
```

Comment:
Since this is a private function (not exported or used outside of this module), rename with an underscore: `def _detect`. Same for `_prune`.